common/src/main/java/org/opensearch/ml/common/model/TextEmbeddingModelConfig.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.model;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import org.opensearch.common.ParseField;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.ml.common.FunctionName;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Setter
+@Getter
+public class TextEmbeddingModelConfig extends MLModelConfig {
+    public static final String PARSE_FIELD_NAME = FunctionName.TEXT_EMBEDDING.name();
+    public static final NamedXContentRegistry.Entry XCONTENT_REGISTRY = new NamedXContentRegistry.Entry(
+            TextEmbeddingModelConfig.class,
+            new ParseField(PARSE_FIELD_NAME),
+            it -> parse(it)
+    );
+
+    public static final String EMBEDDING_DIMENSION_FIELD = "embedding_dimension";
+    public static final String FRAMEWORK_TYPE_FIELD = "framework_type";
+
+    private Integer embeddingDimension;
+    private FrameworkType frameworkType;
+
+    @Builder
+    public TextEmbeddingModelConfig(String modelType, Integer embeddingDimension, FrameworkType frameworkType, String allConfig) {
```

Comment:
can't we use @NonNull?

Reply:
That's possible. I think that's not bad to write few more lines of code to make the error message more readable.